pkg/controller/instance/instance_controller_test.go

```diff
@@ -1,156 +1,20 @@
-	"context"
-	"fmt"
-	"log"
-	"os"
-	"path/filepath"
-	"sync"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/uuid"
-	"k8s.io/client-go/discovery/cached/memory"
-	"k8s.io/client-go/kubernetes/scheme"
-	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 
-	"github.com/kudobuilder/kudo/pkg/apis"
 	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
 	"github.com/kudobuilder/kudo/pkg/engine"
 	"github.com/kudobuilder/kudo/pkg/engine/task"
-	"github.com/kudobuilder/kudo/pkg/kubernetes"
-	"github.com/kudobuilder/kudo/pkg/util/convert"
-	"github.com/kudobuilder/kudo/pkg/util/kudo"
 )
 
-const timeout = time.Second * 5
-const tick = time.Millisecond * 500
-
-var cfg *rest.Config
-
-func TestMain(m *testing.M) {
-	t := &envtest.Environment{
-		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "config", "crds")},
-	}
-
-	if err := apis.AddToScheme(scheme.Scheme); err != nil {
-		log.Fatal(err)
-	}
-
-	var err error
-	if cfg, err = t.Start(); err != nil {
-		log.Fatal(err)
-	}
-
-	code := m.Run()
-	// t.Stop() returns an error, but since we exit in the next line anyway, it is suppressed
-	t.Stop() //nolint:errcheck
-	os.Exit(code)
-}
-
-func TestRestartController(t *testing.T) {
-	stopMgr, mgrStopped, c := startTestManager(t)
-
-	log.Printf("Given an existing instance 'foo-instance' and operator 'foo-operator'")
-	in := &v1beta1.Instance{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo-instance", Namespace: "default", Labels: map[string]string{kudo.OperatorLabel: "foo-operator"}},
-		Spec: v1beta1.InstanceSpec{
-			OperatorVersion: v1.ObjectReference{
-				Name:      "foo-operator",
-				Namespace: "default",
-			},
-			Parameters: map[string]string{"param": "value"},
-		},
-	}
-	instanceKey, _ := client.ObjectKeyFromObject(in)
-	assert.NoError(t, c.Create(context.TODO(), in))
-	defer c.Delete(context.TODO(), in)
-
-	ov := &v1beta1.OperatorVersion{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo-operator", Namespace: "default"},
-		TypeMeta:   metav1.TypeMeta{Kind: "OperatorVersion", APIVersion: "kudo.dev/v1beta1"},
-		Spec: v1beta1.OperatorVersionSpec{
-			Plans: map[string]v1beta1.Plan{"deploy": {Phases: []v1beta1.Phase{}}, "update": {Phases: []v1beta1.Phase{}}},
-			Parameters: []v1beta1.Parameter{
-				{
-					Name:    "param",
-					Default: convert.StringPtr("default"),
-				},
-			},
-		},
-	}
-	assert.NoError(t, c.Create(context.TODO(), ov))
-	defer c.Delete(context.TODO(), ov)
-
-	log.Print("And a deploy plan that was already run")
-	assert.Eventually(t, func() bool { return instancePlanFinished(instanceKey, "deploy", c) }, timeout, tick)
-
-	log.Print("When we stop the manager")
-	close(stopMgr)
-	mgrStopped.Wait()
-
-	log.Print("And update the instance parameter value")
```

Comment:
This test is not needed anymore as the webhook will decline Instance updates when the manager is down.